### PR TITLE
Update Trilinos version. Update configurations and removed unused configuration files

### PR DESCRIPTION
**Pull request by jrood-nrel**

### CODE REVIEW

File: repos/exawind/packages/trilinos/package.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ class Trilinos(bTrilinos, CmakeExtension):
     # the trilinos github page appended with the date of the commit.
     # this preserves the Trilinos versioning scheme and will allow for valid
     # version comparisons in spack's internals.
+    version("15.1.1", tag="trilinos-release-15-1-1")
     version("14.1.0.2023.02.28", commit="8b3e2e1db4c7e07db13225c73057230c4814706f")
```
Comment (psakievich) [inline repos/exawind/packages/trilinos/package.py:21]: move this back to `13.4.1`?

File: configs/base/packages.yaml [hunk 2]
```diff
@@ -24,6 +24,8 @@ packages:
     variants: ~fortran
   trilinos:
     variants: ~adelus~adios2~amesos~amesos2~anasazi~asan+aztec~basker~belos~boost~chaco~complex~dtk~epetra~epetraext~epetraextbtf~epetraextexperimental~epetraextgraphreorderings~float~fortran~hypre~ifpack~ifpack2~intrepid~intrepid2~ipo~isorropia~mesquite~minitensor~ml~muelu~mumps~nox~openmp~panzer~phalanx~piro~python~rol~rythmos~sacado~scorec~shards~shylu~stk_simd~stk_unit_tests~stokhos~stratimikos~strumpack~suite-sparse~superlu~superlu-dist~teko~tempus~test~thyra~trilinoscouplings~x11~zoltan2
+    require:
+      - any_of: ["@=15.1.1", "@=develop"]
```
Comment (psakievich) [inline configs/base/packages.yaml:28]: I think we need to keep the `13.4.1` release as an option as well, but you can put it last so it is the lowest precedent.

If not then we won't be able to build the `exawind@1.0` release.

**Review by psakievich — CHANGES_REQUESTED** [review]

File: repos/exawind/packages/trilinos/package.py (hunk 1 above)
Comment (jrood-nrel) [reply repos/exawind/packages/trilinos/package.py:21]: I think we want to try to use the latest trilinos version with nalu though. I figured we could switch to actual releases.

Comment (psakievich) [reply repos/exawind/packages/trilinos/package.py:21]: I agree with that. I just want to make sure we can build the challenge problem that ran across platforms for historical purposes.

Comment (psakievich) [inline repos/exawind/packages/trilinos/package.py:21]:
```suggestion
    version("13.4.1.2023.02.28", commit="8b3e2e1db4c7e07db13225c73057230c4814706f")
```
This is what I mean.

File: repos/exawind/packages/nalu-wind/package.py [hunk 3]
```diff
@@ -51,8 +51,6 @@ class NaluWind(CmakeExtension, bNaluWind, ROCmPackage):
     depends_on("trilinos gotype=long")
     depends_on("openfast@develop,fsi+netcdf+cxx", when="+fsi")
 
-    patch("rocm1.patch")
```
Comment (psakievich) [inline repos/exawind/packages/nalu-wind/package.py:54]: should this be kept for certain versions?

**Review by psakievich — APPROVED** [review]

File: repos/exawind/packages/nalu-wind/package.py (hunk 3 above)
Comment (jrood-nrel) [reply repos/exawind/packages/nalu-wind/package.py:54]: No, it's been applied in https://github.com/Exawind/nalu-wind/pull/1255

**Pull request merged**